### DISCUSSION TITLE
FFMPEG Copy - Best Practice

### DISCUSSION THREAD

**Discussion by AdiAbuAli** [bluenviron/mediamtx#681] [Questions]
Hi,
Am using wyze bridge by MRLT8, to stream my Wyze Cam v3 via RTSP (using RTSP-Simple-Server), and it stream only Video....

I tried both commands:
**-loglevel error -use_wallclock_as_timestamps 1 -flags +global_header+low_delay -fflags +flush_packets+genpts+nobuffer -i - -vcodec copy -rtsp_transport tcp -f rtsp rtsp://0.0.0.0:8554**

And the result if ok, but live stream is not too smooth & lagging , and the records doesn't copy the stream 100% (around 99.8% down to 50% on Cams with bad wifi coverage ).

Also tried below command:
**-loglevel error -fflags +genpts -i - -c:v libx264 -preset ultrafast -b:v 960k -max_muxing_queue_size 2048 -rtsp_transport tcp -f rtsp rtsp://0.0.0.0:8554**

And the result was perfect, but CPU jumped X4 !

**Am using Synology Surveillance station (4-core 2.0 GHz, burst up to 2.7 GHz)....
Wyze CAM v3 bitrate is 120KB, HD, and FPS=20....**

Any suggestion to tune up FFMPEG command, or thanks to share your experience(s)... ?

**Comment by aler9** [comment]
Hello, try switching to hardware-accelerated encoding, it drastically decreases CPU consumption. You need to compile a custom version of ffmpeg and change the command. I know that it's time consuming but i can guarantee you that it's worth the effort:

https://trac.ffmpeg.org/wiki/HWAccelIntro

**Reply by cothingsusr** [reply]
@aler9 is there a way to publish the RAW stream with ffmpeg copy stream without the need to use libx264 and at the same time keep same video quality?

**Reply by aler9** [reply]
> @aler9 is there a way to publish the RAW stream

There's a [RTP payload format](https://en.wikipedia.org/wiki/RTP_payload_formats) for raw video, but i don't think that ffmpeg supports it, and even if it does, your CPU or network bandwidth would not hold it. I once used [ROS](https://www.ros.org/) for video streaming, it allows to exchange raw video frames, and a single 1080p 25ps stream (1080 x 1920 x 3 x 25 = 155 MB/s) was enough to make the network collapse and the CPU going to 100%.

**Reply by cothingsusr** [reply]
Thanks for your feedback

[a comment by an automated account is not shown]
